staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator.go

```diff
@@ -97,6 +121,48 @@ func (a *cachedTokenAuthenticator) AuthenticateToken(ctx context.Context, token
 	return resp, ok, err
 }
 
-func keyFunc(auds []string, token string) string {
-	return fmt.Sprintf("%#v|%v", auds, token)
+// keyFunc generates a string key by hashing the inputs.
+// This lowers the memory requirement of the cache and keeps tokens out of memory.
+func keyFunc(hashPool *sync.Pool, auds []string, token string) string {
+	h := hashPool.Get().(hash.Hash)
+
+	h.Reset()
+
+	// try to force stack allocation
+	var a [4]byte
```

Comment:
nit: since this doesn't allocate, it's fine to just do it within the function, making it take fewer parameters.

Reply:
Apparently it causes allocations when called from a `for` loop:

```
BenchmarkKeyFunc/has_audiences-4         	 2000000	      4452 ns/op	      48 B/op	       5 allocs/op
BenchmarkKeyFunc/nil_audiences-4         	 2000000	      3842 ns/op	      36 B/op	       2 allocs/op
PASS
```

Reply:
hm, we must be defeating the escape analysis :(